scipy/optimize/tests/test_nonlin.py

```diff
@@ -217,6 +219,41 @@ def wont_converge(x):
         with pytest.raises(scipy.optimize.NoConvergence):
             nonlin.newton_krylov(wont_converge, xin=[0], maxiter=1)
 
+    def test_warnings_invalid_inner_param(self):
+        """
+        Test for ENH #21986, for behavior of `nonlin.newton_krylov`
+        Test the following scenarios:
+        1. Raise warning for invalid inner param
+        2. No warning for valid inner param
+        3. No warning for user-provided callable method
+        """
+
+        # catch all warnings
+        warnings.simplefilter("always")
+
+        # This should raise exactly one warning
+        # (`inner_atol` is not valid for `minres`)
+        with warnings.catch_warnings(record=True) as w:
+            nonlin.newton_krylov(F, F.xin, method="minres", inner_atol=1e-5)
+        assert_(len(w) == 1)
```

Comment:
We prefer to use the builtin pytest `assert` statements now for new tests

```suggestion
        assert len(w) == 1
```